src/tuxclocker-qt/MainWindow.cpp

```diff
@@ -104,27 +104,34 @@ MainWindow::MainWindow(QWidget *parent) : QMainWindow(parent) {
 
 void MainWindow::setTrayIconEnabled(bool enable) {
 	if (enable) {
-		if (!m_trayIcon)
-			// This seems to make the main window not close during closeEvent
+		if (!m_trayIcon) {
 			m_trayIcon = new QSystemTrayIcon{this};
-		m_trayIcon->setIcon(QIcon{QPixmap{":/tuxclocker-logo.svg"}});
-		m_trayIcon->setToolTip("TuxClocker");
-		m_trayIcon->setContextMenu(createTrayMenu());
-		m_trayIcon->show();
-		return;
+			m_trayIcon->setIcon(QIcon{QPixmap{":/tuxclocker-logo.svg"}});
+			m_trayIcon->setToolTip("TuxClocker");
+			m_trayIcon->setContextMenu(createTrayMenu());
+			m_trayIcon->show();
+			return;
+		} else
+			return;
 	}
-	// Remove tray icon
-	if (m_trayIcon)
+
+	if (m_trayIcon) {
 		delete m_trayIcon;
+		m_trayIcon = nullptr;
+	}
 }
 
 QMenu *MainWindow::createTrayMenu() {
 	auto menu = new QMenu{this};
 
-	auto show = new QAction{_("&Maximize TuxClocker"), this};
+	auto show = new QAction{_("&Show TuxClocker"), this};
```

Comment:
I don't think this needs changing, I think Maximize/Hide is clear enough, and translations would need to be changed